Quick fix for 0.51.x - upstream has reworked in git master with a more
complete fix:

https://github.com/mesonbuild/meson/commit/3a011df4df1e57536e5792350fb932f0f9eacb0a

From 4194f045c171b17b5f25b811ac734aa3f8dead29 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 7 Sep 2019 21:45:58 -0700
Subject: [PATCH] Disable crossfile tests on Solaris

They all fail because /usr/bin/cc is *not* gcc or clang

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 run_unittests.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/run_unittests.py b/run_unittests.py
index 02991aa6..93572ca4 100755
--- a/run_unittests.py
+++ b/run_unittests.py
@@ -2737,6 +2737,8 @@ int main(int argc, char **argv) {
     def test_cross_file_system_paths(self):
         if is_windows():
             raise unittest.SkipTest('system crossfile paths not defined for Windows (yet)')
+        if is_sunos():
+            raise unittest.SkipTest('system crossfile paths not defined for Solaris (yet)')
 
         testdir = os.path.join(self.common_test_dir, '1 trivial')
         cross_content = textwrap.dedent("""\
@@ -4885,6 +4887,8 @@ class LinuxlikeTests(BasePlatformTests):
         self.run_target('coverage-html')
 
     def test_cross_find_program(self):
+        if is_sunos():
+            raise unittest.SkipTest('system crossfile paths not defined for Solaris (yet)')
         testdir = os.path.join(self.unit_test_dir, '11 cross prog')
         crossfile = tempfile.NamedTemporaryFile(mode='w')
         print(os.path.join(testdir, 'some_cross_tool.py'))
@@ -5229,6 +5233,8 @@ endian = 'little'
         self.build()
 
     def test_identity_cross(self):
+        if is_sunos():
+            raise unittest.SkipTest('system crossfile paths not defined for Solaris (yet)')
         testdir = os.path.join(self.unit_test_dir, '58 identity cross')
         crossfile = tempfile.NamedTemporaryFile(mode='w')
         os.environ['CC'] = '"' + os.path.join(testdir, 'build_wrapper.py') + '"'
-- 
2.15.2

